### PR TITLE
Add test fcr

### PR DESCRIPTION
**Pull request by tutkat**
For the PR were introduced changes:

- changed api client (okhttp on rest-assured) and regenerated all classes
- modified existing tests and adjusted for new client
- added tests for Cloud Router endpoints
- added test for Port endpoint
- added flow to run tests only

### CODE REVIEW

**Comment by thogarty** [comment]
@tutkat , please add descriptions of why you changed what you changed. The status update was confusing as well so it's important to add as much context as you can to whomever will be reviewing your code.

File: templates/Java/libraries/okhttp-gson/ApiClient.mustache [hunk 1]
```diff
@@ -1,1885 +0,0 @@
...
-
-package {{invokerPackage}};
-
-{{#dynamicOperations}}
-import io.swagger.v3.oas.models.OpenAPI;
-import io.swagger.v3.oas.models.Operation;
-import io.swagger.v3.oas.models.PathItem;
-import io.swagger.v3.oas.models.parameters.Parameter;
-import io.swagger.v3.oas.models.parameters.Parameter.StyleEnum;
-import io.swagger.v3.parser.OpenAPIV3Parser;
```
Comment (thogarty) [inline templates/Java/libraries/okhttp-gson/ApiClient.mustache:11]: There was a change in this template to modify the default timeouts returned by the API Client. You'll need to maintain that same default in the rest-assured API client.

**Review by thogarty — COMMENTED** [review]
It looks like this is a breaking change for existing users of the Java SDK. What will be the outcome to the user? Do they have to adjust their code based on this update?

We need to prevent breaking changes because our customers have to make many modifications as a result. We only make them if absolutely necessary and I still haven't seen enough evidence in your description to warrant the breaking change. What was the exact issue that okhttp-gson gave you that rest-assured is not?

File: .github/workflows/maven-publish.yml [hunk 2]
```diff
@@ -19,6 +19,7 @@ jobs:
 
     env:
       TEST_DATA_UAT_QINQ_PORT: ${{ secrets.TEST_DATA_UAT_QINQ_PORT }}
+      TEST_DATA_UAT_FCR_USER: ${{secrets.TEST_DATA_UAT_FCR_USER }}
```
[the remark this answers is not in the document]
Comment (thogarty) [reply .github/workflows/maven-publish.yml:22]: This needs to be changed.

**Review by thogarty — CHANGES_REQUESTED** [review]
Workflow needs to be modified based on comments.

**Review by thogarty — CHANGES_REQUESTED** [review]
This is absolutely a breaking change requiring customer updates. Please provide ample evidence for this adjustment.

**Comment by tutkat** [comment]
> It looks like this is a breaking change for existing users of the Java SDK. What will be the outcome to the user? Do they have to adjust their code based on this update?
> 
> We need to prevent breaking changes because our customers have to make many modifications as a result. We only make them if absolutely necessary and I still haven't seen enough evidence in your description to warrant the breaking change. What was the exact issue that okhttp-gson gave you that rest-assured is not?

I don't know the reason but this client wasn't able to execute POST calls. I'm sure any client didn't do that because someone reports that doesn't work. I spent few days to try to configure it but unfortunately I was forced to move to another http client

File: v4/api/ConnectionsApiTest.java [hunk 3]
```diff
@@ -11,140 +10,308 @@
...
-     *
-     * @throws ApiException if the Api call fails
-     */
     @Test
-    public void createConnectionTest() throws ApiException {
-        //
-        //ConnectionPostRequest connectionPostRequest = null;
-        //
-        //Connection response = api.createConnection(connectionPostRequest);
+    public void createConnectionPort2SP() {
```
[the remark this answers is not in the document]
Comment (tutkat) [reply v4/api/ConnectionsApiTest.java:50]: I gave answered in other topic

File: templates/Java/libraries/okhttp-gson/ApiClient.mustache (hunk 1 above)
Comment (tutkat) [reply templates/Java/libraries/okhttp-gson/ApiClient.mustache:11]: what "default timeouts" you mean?

**Comment by ctreatma** [comment]
> I don't know the reason but this client wasn't able to execute POST calls.

How does this issue manifest?  Are the requests never sent at all?  Are they sent in an invalid format and failing?  Is the API responding with an HTTP error?  Are all POST calls impacted or only for particular endpoints?  The metal-java SDK also uses okhttp-gson, and although metal-java is now defunct it was definitely used successfully to send POST requests, which indicates it should be possible for fabric-java to send POST requests successfully without introducing a breaking change to the client.

**Comment by thogarty** [comment]
@tutkat, please remove code generation from this PR. You need to figure out how to separate your test files from any changes you've made to the SDK during this time. There shouldn't be changes to 1,209 files just to add your tests.

**Comment by thogarty** [comment]
@tutkat , this PR cannot be reviewed until it has been trimmed.

**Pull request closed without merging**